<a href="https://colab.research.google.com/github/DAGP1145/Mineriadedatos/blob/Control-de-cambios/Mineria_De_Datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Fuente: https://exoplanetarchive.ipac.caltech.edu/index.html

Disponemos de muchas fuentes de informaciòn de las cuales teniamos dos principales opciones:

Sistema planetarios o sistemas planetarios compuestos.

La diferencia es que uno esta compuesto de un estudio en especifico por lo tanto lo datos son consistentes. Pero la desventaja es que podrian faltar datos porque simplemente ese estudio no midiò todo.

Por otro lado la segunda tabla esta compuesta de distintas publicaciones y las junta para crear una sola fila por planeta, por lo tanto es mas completa. El problema es que esa información puede no ser consistente, porque viene de distintos estudios que usaron métodos diferentes.

Se seleccionó la tabla de Parámetros Compuestos, ya que proporciona una única fila por planeta con un conjunto de datos más completo, lo que facilita el análisis y la aplicación de técnicas de minería de datos, aun cuando los datos no sean completamente consistentes entre sí.

# Descripción de las variables seleccionadas

## Variables Cuantitativas Discretas

| Variable    | Descripción                                    |
|:------------|:-----------------------------------------------|
| `sy_snum`     | Número de estrellas en el sistema planetario   |
| `sy_pnum`     | Número de planetas en el sistema               |
| `sy_mnum`     | Número de lunas en el sistema                  |
| `disc_year`   | Año en que se descubrió el planeta             |

## Variables Cuantitativas Continuas

| Variable      | Descripción                                                                 |
|:--------------|:----------------------------------------------------------------------------|
| `pl_orbper`   | Período orbital del planeta medido en días terrestres                       |
| `pl_dens`     | Densidad media del planeta                                                  |
| `pl_rade`     | Radio del planeta medido en radios terrestres                               |
| `pl_radj`     | Radio del planeta medido en radios de Júpiter                               |
| `pl_orbeccen` | Excentricidad orbital (qué tan elíptica es la órbita)                     |
| `pl_eqt`      | Temperatura de equilibrio del planeta                                       |
| `st_teff`     | Temperatura efectiva de la estrella                                         |
| `st_rad`      | Radio de la estrella, medido en unidades de radios solares                  |
| `st_mass`     | Masa de la estrella, medida en masas solares                                |
| `st_met`      | Metalicidad de la estrella (proporción de metales)                          |
| `st_logg`     | Gravedad superficial de la estrella                                         |
| `sy_dist`     | Distancia al sistema planetario medida en pársecs                           |

## Variables Cualitativas

### Variables Cualitativas Nominales

| Variable          | Descripción                                                                       |
|:------------------|:----------------------------------------------------------------------------------|
| `pl_name`         | Nombre del planeta                                                                |
| `hostname`        | Nombre de la estrella anfitriona                                                  |
| `discoverymethod` | Método de descubrimiento del planeta                                              |
| `disc_facility`   | Lugar de observación del descubrimiento del planeta (Tierra o Espacio)           |
| `disc_telescope`  | Nombre del telescopio utilizado                                                   |
| `disc_instrument` | Nombre del instrumento de observación para el descubrimiento de planetas           |

### Variables Cualitativas Ordinales

| Variable      | Descripción                                                                     |
|:--------------|:--------------------------------------------------------------------------------|
| `st_spectype` | Clasificación espectral de la estrella (sistema Morgan-Keenan)                  |


# Importación de librerías

In [2]:
import pandas as pd # Estructura y manipulación de datos
import numpy as np #Programación vectorial

# Carga de los datos



In [3]:
url = 'https://raw.githubusercontent.com/DAGP1145/Mineriadedatos/refs/heads/main/PSCompPars_2026.04.21_08.17.48.csv'
df = pd.read_csv(url, sep=",", comment="#")

#Exploración inicial de los datos

In [40]:
display(df.head())

,pl_name,hostname,sy_snum,sy_pnum,sy_mnum,discoverymethod,disc_year,disc_locale,disc_facility,disc_telescope,...,st_loggerr1,st_loggerr2,st_logglim,st_dens,st_denserr1,st_denserr2,st_denslim,sy_dist,sy_disterr1,sy_disterr2
0,11 Com b,11 Com,2,1,0,Radial Velocity,2007.0,Ground,Xinglong Station,2.16 m Telescope,...,0.08,-0.08,0.0,NaN,NaN,NaN,NaN,93.1846,1.9238,-1.9238
1,11 UMi b,11 UMi,1,1,0,Radial Velocity,2009.0,Ground,Thueringer Landessternwarte Tautenburg,2.0 m Alfred Jensch Telescope,...,0.07,-0.07,0.0,NaN,NaN,NaN,NaN,125.3210,1.9765,-1.9765
2,14 And b,14 And,1,1,0,Radial Velocity,2008.0,Ground,Okayama Astrophysical Observatory,1.88 m Telescope,...,0.06,-0.07,0.0,NaN,NaN,NaN,NaN,75.4392,0.7140,-0.7140
3,14 Her b,14 Her,1,2,0,Radial Velocity,2002.0,Ground,W. M. Keck Observatory,10 m Keck I Telescope,...,0.02,-0.02,0.0,1.273928,0.349533,-0.30728,0.0,17.9323,0.0073,-0.0073
4,16 Cyg B b,16 Cyg B,3,1,0,Radial Velocity,1996.0,Ground,Multiple Observatories,Multiple Telescopes,...,0.01,-0.01,0.0,1.011029,0.292731,-0.20081,0.0,21.1397,0.0110,-0.0111


In [5]:
display(df.tail())

,pl_name,hostname,sy_snum,sy_pnum,sy_mnum,discoverymethod,disc_year,disc_locale,disc_facility,disc_telescope,...,st_loggerr1,st_loggerr2,st_logglim,st_dens,st_denserr1,st_denserr2,st_denslim,sy_dist,sy_disterr1,sy_disterr2
6155,ups And b,ups And,2,3,0,Radial Velocity,1996.0,Ground,Lick Observatory,3.0 m C. Donald Shane Telescope,...,0.024844,-0.024844,0.0,0.36993,0.099225,-0.071557,0.0,13.4054,0.06350,-0.06290
6156,ups And c,ups And,2,3,0,Radial Velocity,1999.0,Ground,Multiple Observatories,Multiple Telescopes,...,0.024844,-0.024844,0.0,0.36993,0.099225,-0.071557,0.0,13.4054,0.06350,-0.06290
6157,ups And d,ups And,2,3,0,Radial Velocity,1999.0,Ground,Multiple Observatories,Multiple Telescopes,...,0.024844,-0.024844,0.0,0.36993,0.099225,-0.071557,0.0,13.4054,0.06350,-0.06290
6158,ups Leo b,ups Leo,1,1,0,Radial Velocity,2021.0,Ground,Okayama Astrophysical Observatory,1.88 m Telescope,...,0.090000,-0.090000,0.0,NaN,NaN,NaN,NaN,52.5973,0.92720,-0.89630
6159,xi Aql b,xi Aql,1,1,0,Radial Velocity,2007.0,Ground,Okayama Astrophysical Observatory,1.88 m Telescope,...,0.080000,-0.080000,0.0,NaN,NaN,NaN,NaN,56.1858,0.55975,-0.55975


La pagina web importa los datos bajo estos criterios:

1.   Elige el valor por defecto entre las distintas publicaciones de un mismo valor si fue definido.

2.   Si no existe, elige el valor con menor incertidumbre entre las distintas trasacciones de un mismo dato.

> Se mide como un rango alrededor del valor (“± algo”) por ejemplo: 10 ± 2 significa que el valor real puede estar entre 8 y 12.

3.  Si hay empate en la incertidumbre, elige el valor de la publicación más reciente

La base de datos te entrega por defecto datos que ya fueron seleccionados automáticamente para ser el más útil, pero no necesariamente el más consistente entre sí. Pero para efectos practicos no sera necesario tomar esta condición como una desventaja. A continuación seleccionaremos las columnas que queremos utilizar, esto lo hacemos para tener una base de datos más simplificada
al Dataframe original.

Como primera observación. Notaremos que en nuestra base de datos existen columnas adicionales con nombres similares, pero con sufijos numéricos. Esto se debe a que representan la incertidumbre del valor principal. Pero para efectos practicos solo se va a utilizar el valor sin numero el numero de referencia.

Ejemplo:

st_dens: es el dato que representa la densidad del sol pero para ese mismo dato existen columnas como:

Ejemplo: st_denserr1 correspone al límite superior del margen de error y st_denserr2 corresponde al límite inferior del margen de error.

In [6]:
systemplanets = df[['pl_name', 'hostname', 'sy_snum', 'sy_pnum', 'sy_mnum', 'discoverymethod', 'disc_year', 'disc_facility', 'disc_telescope', 'disc_instrument', 'pl_orbper', 'pl_rade', 'pl_radj', 'pl_dens', 'pl_orbeccen', 'pl_eqt', 'st_spectype', 'st_teff', 'st_rad', 'st_mass', 'st_met', 'st_logg', 'sy_dist']]

## Exploración básica de los datos

## Enfoque del Problema: Agrupación de Exoplanetas

Dado que nuestro objetivo es explorar los datos y aprender acerca de astronomía, hemos pensado que nuestra investigación este orientada a la **Agrupación de Exoplanetas** como la problemática principal.

El objetivo de este enfoque es identificar patrones y estructuras naturales dentro del conjunto de datos de exoplanetas, agrupando aquellos que comparten características similares. Esto nos permitirá:

*   **Descubrir tipos de exoplanetas:** Identificar si existen categorías o clases intrínsecas de exoplanetas basadas en sus propiedades físicas (radio, densidad, temperatura, etc.) y orbitales (período orbital, excentricidad).
*   **Exploración sin supervisión:** Al no requerir una variable objetivo predefinida, el clustering es ideal para una fase exploratoria, donde buscamos entender la variabilidad y la organización inherente de los datos.
*   **Identificación de puntos de interés:** Los grupos identificados pueden señalar características o patrones que merezcan una investigación más profunda, sugiriendo preguntas para futuros análisis.

Este proceso nos ayudará a familiarizarnos con el dataset de una manera intuitiva y visual, sentando las bases para análisis más complejos en el futuro.

In [7]:
display(systemplanets.sample())

,pl_name,hostname,sy_snum,sy_pnum,sy_mnum,discoverymethod,disc_year,disc_facility,disc_telescope,disc_instrument,...,pl_dens,pl_orbeccen,pl_eqt,st_spectype,st_teff,st_rad,st_mass,st_met,st_logg,sy_dist
527,HD 10180 e,HD 10180,1,6,0,Radial Velocity,2010.0,La Silla Observatory,3.6 m ESO Telescope,HARPS Spectrograph,...,0.881,0.051,NaN,G1 V,5911.0,1.109,1.06,0.08,4.39,38.9607


In [8]:
# df.shape para conocer las dimensiones de la base de datos.
systemplanets.shape

(6160, 23)

In [54]:
print('Numero total de columnas', sum(systemplanets.columns.value_counts()),'\n')
display(systemplanets.columns.to_list())

Numero total de columnas 23 



['pl_name',
 'hostname',
 'sy_snum',
 'sy_pnum',
 'sy_mnum',
 'discoverymethod',
 'disc_year',
 'disc_facility',
 'disc_telescope',
 'disc_instrument',
 'pl_orbper',
 'pl_rade',
 'pl_radj',
 'pl_dens',
 'pl_orbeccen',
 'pl_eqt',
 'st_spectype',
 'st_teff',
 'st_rad',
 'st_mass',
 'st_met',
 'st_logg',
 'sy_dist']

In [10]:
#Pero no visualizamos cuáles son las columnas que en su conjunto pertenecen a cierto tipo de dato. A continuación lo hacemos.
df_objetos = systemplanets.select_dtypes(include=['object'])
df_int_floats = systemplanets.select_dtypes(include=['int', 'float'])

In [67]:
display(df_objetos.columns.to_list())
print("Numero de columnas: ", sum(df_objetos.columns.value_counts()), '\n')
display(df_int_floats.columns.to_list())
print("Numero de columnas: ", sum(df_int_floats.columns.value_counts()))

['pl_name',
 'hostname',
 'discoverymethod',
 'disc_facility',
 'disc_telescope',
 'disc_instrument',
 'st_spectype']

Numero de columnas:  7 



['sy_snum',
 'sy_pnum',
 'sy_mnum',
 'disc_year',
 'pl_orbper',
 'pl_rade',
 'pl_radj',
 'pl_dens',
 'pl_orbeccen',
 'pl_eqt',
 'st_teff',
 'st_rad',
 'st_mass',
 'st_met',
 'st_logg',
 'sy_dist']

Numero de columnas:  16


In [9]:
# Con info() Podremos visualizar las columnas, los tipos de datos y su cantidad. Además podremos tener algun indicio de si existe nulos en el dataset.
systemplanets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6160 entries, 0 to 6159
Data columns (total 23 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   pl_name          6160 non-null   object 
 1   hostname         6160 non-null   object 
 2   sy_snum          6160 non-null   int64  
 3   sy_pnum          6160 non-null   int64  
 4   sy_mnum          6160 non-null   int64  
 5   discoverymethod  6160 non-null   object 
 6   disc_year        6159 non-null   float64
 7   disc_facility    6160 non-null   object 
 8   disc_telescope   6160 non-null   object 
 9   disc_instrument  6160 non-null   object 
 10  pl_orbper        5821 non-null   float64
 11  pl_rade          6110 non-null   float64
 12  pl_radj          6110 non-null   float64
 13  pl_dens          6021 non-null   float64
 14  pl_orbeccen      5221 non-null   float64
 15  pl_eqt           4596 non-null   float64
 16  st_spectype      2314 non-null   object 
 17  st_teff       

Observación: En la columna Non-null Count se observan variaciones en la cantidad de datos disponibles entre los distintos atributos, lo que indica la presencia de valores nulos en el dataset. A continuación, se presenta un listado de dichos atributos.

 |disc_year       |6159 non-null    |float64    |
|:----------------|:----------------|:---------|
|  `pl_orbper`    |  5821 non-null  |  float64
|  `pl_rade`      |  6110 non-null  |  float64
|  `pl_radj`      |  6110 non-null  |  float64
|  `pl_dens`      |  6021 non-null  |  float64
|  `pl_orbeccen`  |  5221 non-null  |  float64
|  `pl_eqt`       |  4596 non-null  |  float64
|  `st_spectype`  |  2314 non-null  |  object
|  `st_teff`      |  5866 non-null  |  float64
|  `st_rad`       |  5842 non-null  |  float64
|  `st_mass`      |  6152 non-null  |  float64
|  `st_met`       |  5606 non-null  |  float64
|  `st_logg`      |  5838 non-null  |  float64
|  `sy_dist`      |  6133 non-null  |  float64


In [60]:
systemplanets.describe().round(2)

,sy_snum,sy_pnum,sy_mnum,disc_year,pl_orbper,pl_rade,pl_radj,pl_dens,pl_orbeccen,pl_eqt,st_teff,st_rad,st_mass,st_met,st_logg,sy_dist
count,6160.00,6160.00,6160.0,6159.00,5.821000e+03,6110.00,6110.00,6021.00,5221.00,4596.00,5866.00,5842.00,6152.00,5606.00,5838.00,6133.00
mean,1.10,1.77,0.0,2017.00,7.353095e+04,5.85,0.52,4.90,0.08,913.31,5396.94,1.50,0.94,0.02,4.38,714.70
std,0.34,1.16,0.0,4.98,5.271453e+06,5.44,0.49,34.52,0.15,464.49,1747.19,3.95,0.41,0.19,0.44,1221.08
min,1.00,1.00,0.0,1992.00,9.000000e-02,0.31,0.03,0.01,0.00,34.00,415.00,0.01,0.01,-1.00,0.54,1.30
25%,1.00,1.00,0.0,2014.00,4.340000e+00,1.83,0.16,1.28,0.00,566.00,4901.50,0.77,0.77,-0.08,4.29,100.95
50%,1.00,1.00,0.0,2016.00,1.110000e+01,2.85,0.25,2.56,0.00,818.00,5545.00,0.95,0.94,0.02,4.45,375.67
75%,1.00,2.00,0.0,2021.00,3.964000e+01,11.99,1.07,4.59,0.09,1164.93,5900.00,1.25,1.09,0.13,4.58,835.86
max,4.00,8.00,0.0,2026.00,4.020000e+08,87.21,7.78,2000.00,0.95,4050.00,57000.00,88.48,10.94,0.79,8.07,8500.00


sy_snum (Numero de estrellas)
1. Promedio es de: 1.10
2. Desviación estándar es de: 0.34
3. Valor mínimo es de: 1.00
4. Valor máximo es de: 4.00
5. Coeficiente de Variación (CV): 0.31

 sy_pnum (Numero de planetas)
1. Promedio es de: 1.77
2. Desviación estándar es de: 1.16
3. Valor mínimo es de: 1.00
4. Valor máximo es de: 8.00
5. Coeficiente de Variación (CV): 0.65

 sy_mnum (Numero de lunas)
1. Promedio es de: 0.00
2. Desviación estándar es de: 0.00
3. Valor mínimo es de: 0.00
4. Valor máximo es de: 0.00
5. Coeficiente de Variación (CV): Indefinido (promedio es cero)

 disc_year (Año en que se descubrió el planeta)
1. Promedio es de: 2017.00
2. Desviación estándar es de: 4.98
3. Valor mínimo es de: 1992.00
4. Valor máximo es de: 2026.00
5. Coeficiente de Variación (CV): 0.00

 pl_orbper (Período orbital del planeta medido en días terrestres)
1. Promedio es de: 73530.95
2. Desviación estándar es de: 5271453.00
3. Valor mínimo es de: 0.09
4. Valor máximo es de: 402000000.00
5. Coeficiente de Variación (CV): 71.69

 pl_rade (Radio del planeta medido en radios terrestres)
1. Promedio es de: 5.85
2. Desviación estándar es de: 5.44
3. Valor mínimo es de: 0.31
4. Valor máximo es de: 87.21
5. Coeficiente de Variación (CV): 0.93

 pl_radj (Radio del planeta medido en radios de Júpiter)
1. Promedio es de: 0.52
2. Desviación estándar es de: 0.49
3. Valor mínimo es de: 0.03
4. Valor máximo es de: 7.78
5. Coeficiente de Variación (CV): 0.94

 pl_dens (Densidad media del planeta)
1. Promedio es de: 4.90
2. Desviación estándar es de: 34.52
3. Valor mínimo es de: 0.01
4. Valor máximo es de: 2000.00
5. Coeficiente de Variación (CV): 7.04

 pl_orbeccen (Excentricidad orbital)
1. Promedio es de: 0.08
2. Desviación estándar es de: 0.15
3. Valor mínimo es de: 0.00
4. Valor máximo es de: 0.95
5. Coeficiente de Variación (CV): 1.88

 pl_eqt (Temperatura de equilibrio del planeta)
1. Promedio es de: 913.31
2. Desviación estándar es de: 464.49
3. Valor mínimo es de: 34.00
4. Valor máximo es de: 4050.00
5. Coeficiente de Variación (CV): 0.51

 st_teff (Temperatura efectiva de la estrella)
1. Promedio es de: 5396.94
2. Desviación estándar es de: 1747.19
3. Valor mínimo es de: 415.00
4. Valor máximo es de: 57000.00
5. Coeficiente de Variación (CV): 0.32

 st_rad (Radio de la estrella, medido en unidades de radios solares)
1. Promedio es de: 1.50
2. Desviación estándar es de: 3.95
3. Valor mínimo es de: 0.01
4. Valor máximo es de: 88.48
5. Coeficiente de Variación (CV): 2.63

 st_mass (Masa de la estrella, medida en masas solares)
1. Promedio es de: 0.94
2. Desviación estándar es de: 0.41
3. Valor mínimo es de: 0.01
4. Valor máximo es de: 10.94
5. Coeficiente de Variación (CV): 0.44

 st_met (Metalicidad de la estrella)
1. Promedio es de: 0.02
2. Desviación estándar es de: 0.19
3. Valor mínimo es de: -1.00
4. Valor máximo es de: 0.79
5. Coeficiente de Variación (CV): 9.50

 st_logg (Gravedad superficial de la estrella)
1. Promedio es de: 4.38
2. Desviación estándar es de: 0.44
3. Valor mínimo es de: 0.54
4. Valor máximo es de: 8.07
5. Coeficiente de Variación (CV): 0.10

 sy_dist (Distancia al sistema planetario medida en pársecs)
1. Promedio es de: 714.70
2. Desviación estándar es de: 1221.08
3. Valor mínimo es de: 1.30
4. Valor máximo es de: 8500.00
5. Coeficiente de Variación (CV): 1.71

Con este resumen tendremos una primera impresión de qué es lo que podriamos esperar observar en la sección de visualización de los datos.

# Limpieza y Preprocesamiento de Datos
## Identificación de valores Nulos

In [55]:
df.isnull().any().any()
# Dio como resultado np.True_. Por lo que si hay valores nulos en alguna parte del DataFrame.

np.True_

In [61]:
#Ubicación de las columnas que contienen nulos.
for feature in systemplanets.columns:
  print('Total de valores nulos de', feature, '=', systemplanets[feature].isna().sum())

Total de valores nulos de pl_name = 0
Total de valores nulos de hostname = 0
Total de valores nulos de sy_snum = 0
Total de valores nulos de sy_pnum = 0
Total de valores nulos de sy_mnum = 0
Total de valores nulos de discoverymethod = 0
Total de valores nulos de disc_year = 1
Total de valores nulos de disc_facility = 0
Total de valores nulos de disc_telescope = 0
Total de valores nulos de disc_instrument = 0
Total de valores nulos de pl_orbper = 339
Total de valores nulos de pl_rade = 50
Total de valores nulos de pl_radj = 50
Total de valores nulos de pl_dens = 139
Total de valores nulos de pl_orbeccen = 939
Total de valores nulos de pl_eqt = 1564
Total de valores nulos de st_spectype = 3846
Total de valores nulos de st_teff = 294
Total de valores nulos de st_rad = 318
Total de valores nulos de st_mass = 8
Total de valores nulos de st_met = 554
Total de valores nulos de st_logg = 322
Total de valores nulos de sy_dist = 27


In [72]:
nulos = systemplanets.isnull().sum()
porcentaje = (nulos / len(df) * 100).round(1)
df_draft = pd.DataFrame({"Nulos": nulos, "Porcentaje": porcentaje})
display(df_draft[df_draft["Nulos"] > 0].sort_values("Porcentaje", ascending = False))

,Nulos,Porcentaje
st_spectype,3846,62.4
pl_eqt,1564,25.4
pl_orbeccen,939,15.2
st_met,554,9.0
pl_orbper,339,5.5
st_rad,318,5.2
st_logg,322,5.2
st_teff,294,4.8
pl_dens,139,2.3
pl_radj,50,0.8


## Limpieza

A continuación, se realizará una limpieza de datos eliminando columnas sin valor analítico.

Mediante la función describe(), se identificaron columnas con valores constantes. Por ejemplo, `sy_mnum` (Numero de lunas) con todos sus valores en cero. Esto genera estadísticas sin variación. Por ello, estas columnas no aportan información relevante y serán eliminadas.

Por otro lado, dado que nuestro enfoque es agrupar los exoplanetas para descubrir e identificar patrones además de estructuras naturales; no sera necesario tener algunos atributos como `disc_facility`, `disc_telescope`, `disc_instrument`, `disc_year`

In [100]:
systemplanets['sy_mnum'].sum()
# El resultado 0 indica que esta columna no registra lunas.

np.int64(0)

In [99]:
#Eliminando columnas sin valor analítico
columnas_sin_valor_analitico = ['sy_mnum', 'disc_facility', 'disc_telescope', 'disc_instrument', 'disc_year']
systemplanets_draft = systemplanets.drop(columns=columnas_sin_valor_analitico)
print(systemplanets_draft.shape)

(6160, 18)


In [101]:
#Ubicación de las columnas que contienen nulos.
for feature in systemplanets_draft.columns:
  print('Total de valores nulos de', feature, '=', systemplanets_draft[feature].isna().sum())

Total de valores nulos de pl_name = 0
Total de valores nulos de hostname = 0
Total de valores nulos de sy_snum = 0
Total de valores nulos de sy_pnum = 0
Total de valores nulos de discoverymethod = 0
Total de valores nulos de pl_orbper = 339
Total de valores nulos de pl_rade = 50
Total de valores nulos de pl_radj = 50
Total de valores nulos de pl_dens = 139
Total de valores nulos de pl_orbeccen = 939
Total de valores nulos de pl_eqt = 1564
Total de valores nulos de st_spectype = 3846
Total de valores nulos de st_teff = 294
Total de valores nulos de st_rad = 318
Total de valores nulos de st_mass = 8
Total de valores nulos de st_met = 554
Total de valores nulos de st_logg = 322
Total de valores nulos de sy_dist = 27


In [147]:
# Cantidad columnas numericas
df_int_floats = systemplanets_draft.select_dtypes(include=['int', 'float'])

In [148]:
display(df_int_floats.columns.to_list())
print("Numero de columnas: ", sum(df_int_floats.columns.value_counts()))

['sy_snum',
 'sy_pnum',
 'pl_orbper',
 'pl_rade',
 'pl_radj',
 'pl_dens',
 'pl_orbeccen',
 'pl_eqt',
 'st_teff',
 'st_rad',
 'st_mass',
 'st_met',
 'st_logg',
 'sy_dist']

Numero de columnas:  14


In [145]:
# Contar el número de valores nulos por fila
nulos_por_fila = systemplanets_draft.isnull().sum(axis=1)

# Filtrar el DataFrame para mostrar solo las filas con más de un valor nulo
filas_con_dos_o_mas_nulos = systemplanets_draft[nulos_por_fila > 1]

display("Filas con más de un valor nulo:", filas_con_dos_o_mas_nulos)
display("Número de filas con más de un valor nulo:", len(filas_con_dos_o_mas_nulos))

'Filas con más de un valor nulo:'

,pl_name,hostname,sy_snum,sy_pnum,discoverymethod,pl_orbper,pl_rade,pl_radj,pl_dens,pl_orbeccen,pl_eqt,st_spectype,st_teff,st_rad,st_mass,st_met,st_logg,sy_dist
7,1RXS J160929.1-210524 b,1RXS J160929.1-210524,1,1,Imaging,NaN,18.647000,1.664,2.54,NaN,1700.0,K7 V,4060.0,1.31316,0.850,NaN,3.99592,139.135000
11,2M0437 b,2MASS J04372171+2651014,1,1,Imaging,NaN,13.000000,1.160,3.18,NaN,1450.0,M V,3100.0,0.84000,0.170,0.01,4.50000,128.484000
12,2MASS J01033563-5515561 AB b,2MASS J01033563-5515561 A,2,1,Imaging,NaN,12.300000,1.100,12.20,NaN,NaN,M5.5/M6,NaN,NaN,0.190,NaN,NaN,47.200000
13,2MASS J01225093-2439505 b,2MASS J01225093-2439505,1,1,Imaging,NaN,11.209000,1.000,30.40,NaN,1600.0,M3.5 V,3530.0,0.36645,0.400,NaN,4.85804,33.828100
14,2MASS J02192210-3925225 b,2MASS J02192210-3925225,1,1,Imaging,NaN,16.140960,1.440,5.77,NaN,NaN,M6,3064.0,0.28100,0.108,NaN,4.59000,40.089641
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6113,ZTF J1230-2655 b,ZTF J1230-2655,1,1,Transit,2.359777e-01,13.787046,1.230,NaN,NaN,NaN,NaN,10000.0,0.01230,0.646,NaN,8.07000,186.690000
6114,ZTF J1828+2308 b,ZTF J1828+2308,1,1,Transit,1.120067e-01,11.130518,0.993,25.30,NaN,NaN,NaN,15900.0,0.01310,0.610,NaN,8.05000,201.827000
6117,b Cen AB b,b Cen A,2,1,Imaging,1.790000e+06,12.400000,1.110,9.98,0.40,NaN,B2.5 V,NaN,NaN,5.500,NaN,NaN,98.093500
6133,kap And b,kap And,1,1,Imaging,NaN,12.300000,1.100,12.80,NaN,1900.0,B9 IV,10900.0,2.31229,2.600,-0.36,3.78000,50.017700


'Número de filas con más de un valor nulo:'

1136